tests/decodecorpus.c

```diff
@@ -662,6 +674,12 @@ static U32 generateSequences(U32* seed, frame_t* frame, seqStore_t* seqStore,
                           MIN(frame->header.windowSize,
                               (size_t)((BYTE*)srcPtr - (BYTE*)frame->srcStart))) +
                          1;
+                if (info.useDict && (RAND(seed) & 1)) {
+                    /* need to occasionally generate offsets that go past the start */
+                    /* we still need to be within the windowSize however */
```

Comment:
Indeed, if the dictionary is larger than `windowSize`, and the encoder uses this opportunity to create offsets larger than `windowSize`, the match must nonetheless end at of before dictionary limit. It cannot overlap.

If the offset is smaller than `windowSize`, then it can overlap with current buffer.